lib/libLog.lua

```diff
@@ -0,0 +1,378 @@
+--[[
+
+    libLog.lua - darktable lua logging library
+
+    Copyright (C) 2016 Bill Ferguson <wpferguson@gmail.com>.
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+]]
+
+local libLog = {}
+libLog.libdoc = {
+  Sections = {"Name", "Synopsis", "Description", "License", "Copyright"},
+  Name = "libLog - darktable lua logging library",
+  Synopsis = [[local log = require "lib/libLog"]],
+  Description = [[libLog provides a multi-level logging solution for use with
+    the darktable lua scripts.]],
+  License = [[This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.]],
+  Copyright = "Copyright (C) 2016 Bill Ferguson <wpferguson@gmail.com>.",
+  functions = {}
+}
+
+local dt = require "darktable"
+
+-- set the default log levels
+
+libLog.debug = {"DEBUG:", false}
+libLog.info = {"INFO:", false}
+libLog.warn = {"WARN:", false}
+libLog.error = {"ERROR:", true}
+libLog.success = {"SUCCESS:", true}
```

Comment:
I thought about the dictionary thing a couple of days ago, when I changed everything back to just printing to stdout.  At that point I needed to get this out and let my mind clear a bit.  I'll go ahead and do this...